src/main/java/com/google/sps/data/Course.java

```diff
@@ -1,19 +1,40 @@
 package com.google.sps.data;
 
-public class Course {
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.CompositeFilterOperator;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import java.util.*;
+
+public class Course implements Comparable<Course> {
   String name;
   String professor;
   Long units;
   String term;
   String school;
+  Key termKey;
+  Key courseKey;
 
   public Course(
-      String courseName, String professorName, Long numUnits, String termName, String schoolName) {
+      String courseName,
+      String professorName,
+      Long numUnits,
+      String termName,
+      String schoolName,
+      DatastoreService db) {
     name = courseName;
     professor = professorName;
     units = numUnits;
     term = termName;
     school = schoolName;
+    List<Key> keys = findKeys(db, schoolName, courseName, termName, units, professorName);
+    termKey = keys.get(0);
+    courseKey = keys.get(1);
```

Comment:
Dagm and I use the findTermKey and findCourseKey functions in the same way  - we only call them on courses that have been successfully added to Datastore, so we know them (and their keys) exist

Reply:
Should I put some kind of default message or a try/catch statement even if we're certain they'll never error out though (because I'm sure there's probably some kind of scenario in which we'd be wrong haha)

Reply:
Eh I'd just let it fail then. If you're pretty sure it's never wrong, it's better to make it fail catastrophically, that way you know for sure if your assumptions were invalid.